test-python/action.yaml

```diff
@@ -0,0 +1,53 @@
+name: Run unit tests with coverage
+description: Run unit tests and upload coverage data.
+
+inputs:
+  python-versions:
+    description: Python versions to run tests on, separated with spaces
+  target:
+    description: Makefile target for the test run. If changed, coverage will not be uploaded.
+    default: coverage
+  tics-token:
+    description: The authorization token for TIOBE TICS
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Set up uv with caching
+      id: setup-uv
+      uses: astral-sh/setup-uv@v3
+      with:
+        enable-cache: true
+    - name: Run tests
+      shell: bash
+      run: |
+        # shellcheck disable=SC1083 # (thinks a GitHub variable is a literal "}")
+        for UV_PYTHON in ${{ inputs.python-versions }}; do
+          export UV_PYTHON
+          echo "::group::Setup tests on Python ${UV_PYTHON}"
+          make setup-tests
+          echo "::endgroup::"
+          echo "::group::Run tests on Python ${UV_PYTHON}"
+          make "${{ inputs.target }}"
+          echo "::endgroup::"
+        done
```

Comment:
Does it mean, that now all `python` versions will be folded and run in a single job ?
If it breaks e.g. in `3.10`, `3.12` will not run, am I correct?

Reply:
Correct. I can make it run all the Python versions regardless of failures if we want, though.